app/models/user.rb

```diff
@@ -37,6 +37,15 @@ class User < ApplicationRecord
   validates_acceptance_of :terms_and_conditions
   validates_confirmation_of :password, on: :create
   validates :currency, inclusion: { in: ExchangeRateService::CURRENCIES }
+  validate :email_uniqueness
+
+  protected
+
+  def email_uniqueness
+    if User.all.map(&:email).include?(email) && !persisted?
+      errors.add(:email, :taken)
+    end
+  end
```

Comment:
Well the email is confirmed though, so how does this user use someone else's email?